modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/query/ElasticsearchQueryTranslator.java

```diff
@@ -67,185 +67,409 @@ public QueryBuilder translate(Query query) {
 			queryBuilder = QueryBuilders.queryStringQuery(query.toString());
 		}
 
-		if (query.getBoost() != null) {
-			queryBuilder.boost(query.getBoost());
-		}
-
 		return queryBuilder;
 	}
 
 	@Override
 	public QueryBuilder visit(BooleanQuery booleanQuery) {
-		return _booleanQueryTranslator.translate(booleanQuery, this);
+		QueryBuilder queryBuilder = _booleanQueryTranslator.translate(
```

Comment:
I think this works, but could it be simplified so that each visit() would only have something like?:

```
	private QueryBuilder _addBoost(Query query, QueryBuilder queryBuilder) {...}
        
	@Override
	public QueryBuilder visit(BooleanQuery booleanQuery) {
		return _addBoost(booleanQuery,_booleanQueryTranslator.translate(booleanQuery, this));
	}

        ...
```

Reply:
Good improvement